pages/content/amp-dev/documentation/guides-and-tutorials/courses/intermediate-course/accepting-user-input-and-displaying-output.md

```diff
@@ -0,0 +1,236 @@
+---
+$title: Accepting User Input and Displaying Output
+$order: 2
+toc: true
+---
+
+## Building a Newsletter Subscription Form
+
+Studies show that every dollar spent on email marketing corresponds to several dollars more in sales. In order to keep our brand in our customers’ minds, we want to send out a regular newsletter with information about our company and deals for discounted cheese bikes and accessories. To build up our mailing list, we have to add a newsletter subscription form. Our product manager wants the form to be as follows:
+
+- The form should accept the user’s full name and email address.
+
+- Both fields should be required.
+
+- If the form submission fails, an error should be displayed to the user.
+
+- If the form submission succeeds, a welcome message with the user’s name should be displayed.
+
+In a non-AMP page, we might reach for a basic HTML form element to collect our users’ information. AMP provides an [`<amp-form>`]({{g.doc('/content/amp-dev/documentation/components/reference/amp-form.md', locale=doc.locale).url.path}}) component, but this component actually uses the element name "form" as well. So, what’s the difference?
+
+In this section, we’ll explore what a traditional website form is, how AMP forms differ from traditional forms, and how to handle the response from the server when submitting AMP forms.
+
+## Introduction To Forms
```

Comment:
Is it intentional that AMP CORS is not mentioned in this part of the course? Without, the form sample wouldn't work if served from an AMP cache.

Reply:
Good point! We mention CORS in the next course, but perhaps we should move that to this one.
We don't explain it in detail in any case - we just refer the more advanced student to appropriate docs.